src/fswAlgorithms/effectorInterfaces/jointMotionCompensator/jointMotionCompensator.cpp

```diff
@@ -0,0 +1,228 @@
+/*
+ ISC License
+
+ Copyright (c) 2025, Autonomous Vehicle Systems Lab, University of Colorado Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+*/
+
+
+#include "fswAlgorithms/effectorInterfaces/jointMotionCompensator/jointMotionCompensator.h"
+#include <iostream>
+#include <cstring>
+
+void JointMotionCompensator::Reset(uint64_t CurrentSimNanos)
+{
+    // check that required input messages are connected
+    if (!this->massMatrixInMsg.isLinked()) {
+        bskLogger.bskLog(BSK_ERROR, "JointMotionCompensator.massMatrixInMsg was not linked.");
+    }
+    if (!this->reactionForcesInMsg.isLinked()) {
+        bskLogger.bskLog(BSK_ERROR, "JointMotionCompensator.reactionForcesInMsg was not linked.");
+    }
+    if (this->jointTorqueInMsgs.empty()) {
+        bskLogger.bskLog(BSK_ERROR, "JointMotionCompensator.jointTorqueInMsgs vector is empty.");
+    } else {
+        for (size_t i = 0; i < this->jointTorqueInMsgs.size(); ++i) {
+            if (!this->jointTorqueInMsgs[i].isLinked()) {
+                bskLogger.bskLog(BSK_ERROR, "JointMotionCompensator.jointTorqueInMsgs[%zu] was not linked.", i);
+            }
+        }
+    }
+
+    this->treeInfoInitialized = false;
```

Comment:
should numberOfSpacecraft be reset to zero on Reset()?  Not sure, just bringing this up.